pkg/util/objectwatcher/objectwatcher.go

```diff
@@ -42,7 +42,7 @@ import (
 // ObjectWatcher manages operations for object dispatched to member clusters.
 type ObjectWatcher interface {
 	Create(ctx context.Context, clusterName string, desireObj *unstructured.Unstructured) error
-	Update(ctx context.Context, clusterName string, desireObj, clusterObj *unstructured.Unstructured) error
+	Update(ctx context.Context, clusterName string, desireObj, clusterObj *unstructured.Unstructured) (bool, error)
```

Comment:
Shall we add a comment to explain the return value, especially for the new boolean?

Reply:
I made some more changes because I realized that the semantics of this return value are a bit vague.
The boolean value currently means "update was made" instead of "successfully updated". I want the metric to more clearly reflect "whether the update was made" and "whether the update was successful", so I changed this return value to an enumeration value.